flink/flink-1.13/src/main/java/io/tidb/bigdata/flink/connector/source/TiDBSchemaAdapter.java

```diff
@@ -270,15 +272,15 @@ public DataType getPhysicalDataType() {
   }
 
   public void open() {
-    this.dateTimeFormatters = new DateTimeFormatter[physicalFieldNames.length];
-    StringBuilder fieldTimestampFormatBuilder = new StringBuilder(TIMESTAMP_FORMAT_PREFIX);
-    int prefixLen = fieldTimestampFormatBuilder.length();
-    for (int idx = 0; idx < physicalFieldNames.length; idx++) {
-      fieldTimestampFormatBuilder.setLength(prefixLen);
-      fieldTimestampFormatBuilder.append(physicalFieldNames[idx]);
-      Optional<String> format =
-          Optional.ofNullable(properties.get(fieldTimestampFormatBuilder.toString()));
-      dateTimeFormatters[idx] = format.map(DateTimeFormatter::ofPattern).orElse(ISO_LOCAL_DATE);
+    if (ArrayUtils.isEmpty(physicalFieldNames)) {
+      this.dateTimeFormatters = new DateTimeFormatter[0];
+      return;
     }
+    this.dateTimeFormatters = Arrays.stream(physicalFieldNames)
+        .map(fieldName ->
+            Optional.ofNullable(properties.get(TIMESTAMP_FORMAT_PREFIX + fieldName))
+                .orElse(properties.get(CMPT_TIMESTAMP_FORMAT_PREFIX + fieldName)))
```

Comment:
It's better to use orElseGet to avoid unnecessary lookup.

Reply:
I did a simple JMH for this, indeed orElseGet performs significantly better than orElse in this case, thx!

Reply:
By the time that we can eliminate JDK 8 support, we can make it even better.